contents/blog/machine-copy-paste-mcp-intro.mdx

```diff
@@ -0,0 +1,65 @@
+---
+title: "MCP: machine copy/paste"
+date: 2025-05-30
+author:
+  - danilo-campos
+featuredImage: >-
+  https://res.cloudinary.com/dmukukwp6/image/upload/robot_6d2cab1b66.png
+featuredImageType: full
+tags:
+    - Engineering
+    - LLM
+---
+
+It’s been hard to talk about MCP – [model context protocol](https://modelcontextprotocol.io/introduction).
+
+It’s just so abstract. Why would you want this? What does it do?
+
+So let’s break it down into actionable terms.
+
+Our meat is slow, and machines are fast. But even machines need a common interface to get things done – that's why MCP exists.
+
+## Are you down with MCP?
+
+In the bad old days of two years ago, using the latest models for writing code meant manually ferrying text between your editor and a browser tab.
+
+It was tedious, sure, but even worse: it was error-prone. The manual interventions were driven by our imprecise brains and fingers, and it was easy to mis-select the text you were trying to move around.
+
+Today, we have apps like Cursor. Instead of a human manually moving code, a tool reads code into the LLM’s context, plans interventions, and executes edits for you. It does this in a tight loop, even detecting and correcting errors as they’re written.
+
+What used to take minutes can happen in seconds. Automation at its finest.
+
+So: copy-paste is the enemy of a good LLM developer experience. The job for developer tools is to make it unnecessary, and MCP exists to support this goal. Get exactly the right data in and out of an agent session, do it fast, don’t make the developer work.
+
+## Saving labor with MCP
+
+First, make it easy to install an MCP server. You can use an [automated tool](/blog/envoy-wizard-llm-agent) to inject the configuration on the developer’s behalf. Check out [this pull request](https://github.com/PostHog/wizard/pull/48) from our wizard for a look at how that works.
```

Comment:
I feel like the target reader of the piece changes here. 

Before this point, it is a general intro to MCP and useful for everyone. At this point, it only becomes relevant to app/dev tool developers. I assume this is the target audience the whole time. 

We should make this a lot clearer in the intro that we are mostly focused on app/dev tool developers and how they can use MCP to be helpful to end users.

Reply:
Also, if you don't want to use an automated tool to inject the configuration, what do you do? I'm curious about the alternative.